lib/membrane_rtc_engine/endpoints/hls/config/sink_bin_config.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Membrane.RTC.Engine.Endpoint.HLS.SinkBinConfig do
+  @moduledoc """
+  Module representing Membrane.HTTPAdaptiveStream.SinkBin configuration for the HLS endpoint.
+  """
+
+  @typedoc """
+    To read more about config options go to module Membrane.HTTPAdaptiveStream.SinkBin and read options descriptions.
+  """
+
+  alias Membrane.HTTPAdaptiveStream.Manifest
+
+  @type t() :: %__MODULE__{
+          manifest_name: String.t(),
+          manifest_module: (Path.t() -> module),
+          storage: Storage.config_t(),
+          target_window_duration: pos_integer | :infinity,
+          persist?: boolean,
+          mode: :live | :vod,
+          hls_mode: :muxed_av | :separate_av,
+          header_naming_fun: (Manifest.Track.t(), counter :: non_neg_integer() -> String.t()),
+          segment_naming_fun: (Manifest.Track.t() -> String.t()),
+          mp4_parameters_in_band?: boolean
+        }
+
+  defstruct manifest_name: "index",
+            manifest_module: Membrane.HTTPAdaptiveStream.HLS,
+            storage: &__MODULE__.default_storage/1,
```

Comment:
You pass a function as a default `storage` value when it expects `Storage.config_t()` which is the passed function's return value. Currently always overwritten by `output_directory` from `hls_endpoint` options, but worth mentioning, I think.